### ISSUE TITLE
Retrieving result of two instances of the same task running in parllel (RPC backend)

### ISSUE THREAD

**Issue by timeu** [celery/celery#3141]
In case of the RPC backend: 

```python
@celery.task
def add (a,b):
    return a+b
```
Running this from an ipython shell:

```python
res=[]
for i in range(1,3):
   res.append(add.delay(i,i))
```
After both tasks are finished, the RabbitMQ broker has two tasks in the result queue

Calling `res[0].get()` will return the result of the first instance. 
However the call will also remove the second result which causes `res[2].get()` to hang.

I am not sure if this is by design. Switching to a `amqp://` backend solves the issue. 
This happens with Celery 4.0.0rc2 and Celery 3.1.20

**Comment by ask** [comment]
Please try master.  We don't have the resources to also fix bugs in 3.1 at this point.

**Comment by ask** [comment]
This pretty much describes the bug we recently fixed in master, so I'm closing this.
